### DISCUSSION TITLE
[ORM] How to query object by a custom field value pattern (regex)?

### DISCUSSION THREAD

**Discussion by drygdryg** [netbox-community/netbox#6478] [Q&A] [answered]
How can I query an object (Device, DeviceRole, etc.) by it's custom field value pattern from my plugin/script/report?
My Device type role has a custom field `backup_location` of type string. I need to fetch all devices, where `backup_location` value matches some string pattern. 
I can request a device by fully matching the custom field value (for example, where `backup_location == 'cbackup://12345'`):
```python
from dcim.models import Device

device = Device.objects.filter(custom_field_data__contains={'backup_location': 'cbackup://12345'})
```
I need to request a model by custom field value pattern (for example, `cbackup:\/\/\d+`). Can I use [regex lookup](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#std:fieldlookup-regex) for this? Any suggestions. Thanks.

**Comment by drmsoffall** [comment, marked as answer]
How about
```
Device.objects.filter(custom_field_data__backup_location__regex=r'cbackup://\d+')
```
Probably not the most efficient QuerySet ever but it should work!

**Reply by jeremystretch** [reply]
> Probably not the most efficient QuerySet

It's actually not bad: `custom_field_data` is a JSON field on the table, `backup_location` is matching on a key within it, and `regex` is just a specific lookup type. So while it seems complex, it's only querying a single table.

**Reply by drygdryg** [reply]
Thank you a lot!
